src/main/java/com/assignment/students/controller/WebController.java

```diff
@@ -1,85 +1,112 @@
 
 import javax.validation.Valid;
+import java.util.ArrayList;
+import java.util.List;
 
 @Controller
 public class WebController {
 
-    private final ClassService classService;
+    private final CourseService courseService;
     private final StudentService studentService;
+    private final CourseStudentService courseStudentService;
 
-    public WebController(ClassService classService, StudentService studentService) {
-        this.classService = classService;
+    public WebController(CourseService courseService, StudentService studentService, CourseStudentService courseStudentService) {
+        this.courseService = courseService;
         this.studentService = studentService;
+        this.courseStudentService = courseStudentService;
     }
 
     @GetMapping("/v1/students")
     public String studentForm(Model model){
-        model.addAttribute("students",new Student());
-        return"students";
+        model.addAttribute("student",new Student());
+        return"student-form";
     }
 
-    @GetMapping("/v1/classes")
+    @GetMapping("/v1/courses")
     public String courseForm(Model model){
-        model.addAttribute("class",new Class());
-        model.addAttribute("classes", classService.getAllClasses());
+        model.addAttribute("course",new Course());
+        model.addAttribute("courses", courseService.getAllCourses());
         return "course-form";
     }
 
 
-    @GetMapping("/v1/classes/{id}/delete")
+    @GetMapping("/v1/courses/{id}/delete")
     public String courseForm(@PathVariable long id,  Model model){
-        classService.deleteClass(id);
-        model.addAttribute("classes",classService.getAllClasses());
-        return "class-registry";
+        courseService.deleteCourse(id);
+        model.addAttribute("courses", courseService.getAllCourses());
+        return "course-registry";
     }
 
-    @GetMapping("/v1/classes/{id}")
-    public String updateClass(@PathVariable long id, Model model){
-        Class subject = classService.getClassById(id);
-        model.addAttribute("class",subject);
-        return "update-class";
+    @GetMapping("/v1/courses/{courseId}/unenroll/{studentId}")
+    public String unEnrollStudentFromCourse(@PathVariable long courseId, @PathVariable long studentId, Model model){
+        courseStudentService.unEnrollStudentFromCourse(courseId,studentId);
+        getCourseDetails(courseId, model);
+        return "course-details";
     }
 
-    @PostMapping(path = "/v1/classes",params="action=save")
-    public String classSubmit(@Valid Class subject, BindingResult result, Model model){
+    @GetMapping("/v1/courses/{id}/studentsEnrolled")
+    public String getStudentsByCourseId(@PathVariable long id, Model model){
+        getCourseDetails(id, model);
+        return "course-details";
+    }
+
+    private void getCourseDetails(@PathVariable long id, Model model) {
+        Course subject = courseService.getCourseById(id);
+        List<Student> studentsList = new ArrayList<>();
```

Comment:
Just for the fun of it rewrite using functional programming. The Java world is moving to functional more and more.

Reply:
updated to use foreach and method reference